src/components/ThumbnailImage.js

```diff
@@ -54,8 +70,12 @@ class ThumbnailImage extends PureComponent {
         } else {
             thumbnailScreenHeight = Math.round(thumbnailScreenWidth * aspectRatio);
         }
+        return {width: thumbnailScreenWidth, height: thumbnailScreenHeight};
+    }
 
-        this.setState({thumbnailWidth: thumbnailScreenWidth, thumbnailHeight: Math.max(40, thumbnailScreenHeight)});
+    updateImageSize({width, height}) {
+        const {width: thumbnailWidth, height: thumbnailScreenHeight} = this.calculateThumbnailImageSize(width, height);
```

Comment:
Why is this unnecessary work storing `thumbnailScreenWidth` in name of `width` and vice versa for `height` too, and destructuring to a different name?.

Can we just clean up this?

Reply:
Have you seen the diff here? It was named like this before.

Reply:
There was just one function before please check the diff again.

Reply:
Also, the args passed to this function are named `height` and `width`. So the response args must be of different name. Another advantage of a diff name is that we don't have to re-specify them while setting the state.

Reply:
What's your suggestion here? Naming it `width` and `height` will cause conflicts with arg name.

Reply:
I've changed this as you suggested.